src/corehost/cli/deps_format.cpp

```diff
@@ -124,6 +124,20 @@ void deps_json_t::reconcile_libraries_with_targets(
 
 pal::string_t get_own_rid()
 {
+#if defined(__APPLE__)
+    pal::os_moniker_t moniker;
+    if (pal::get_os_moniker(&moniker)) {
+        if (moniker >= os_moniker_t::sierra) {
+            return pal::string_t("osx.10.12-") + get_arch();
+        }
+        if (moniker >= os_moniker_t::el_capitan) {
+            return pal::string_t("osx.10.11-") + get_arch();
+        }
+        if (moniker >= os_moniker_t::yosemite) {
+            return pal::string_t("osx.10.10-") + get_arch();
+        }
+    }
+#endif
```

Comment:
I see - you are trying to do fallback to use build time RID incase getosmoniker fails, is that right?